server/src/main/java/com/google/musicanalysis/site/OAuthLandingServlet.java

```diff
@@ -0,0 +1,25 @@
+package com.google.musicanalysis.site;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** 
+ * New page after logging in via oauth. This servlet closes the page.
+ */
+@WebServlet("/api/oauth/landing")
+public class OAuthLandingServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest req, HttpServletResponse res) throws IOException {
+    res.setContentType("text/html");
+
+    StringBuilder content = new StringBuilder();
```

Comment:
nit: Don't see a need for `StringBuilder` here since you know what the string is already. Also you can put the script tag outside of the body.
```
String content = '<html>'
               + '<body>Successfully logged in.</body>'
               + '<script type="text/javascript">window.close();</script>'
               + '</html>';

res.getWriter().write(content);
```

Reply:
Yeah, you only need to use a `StringBuilder` if the content is based on some variable or you're constructing the string in a loop.